packages/driver/cypress/e2e/e2e/csp-headers.cy.js

```diff
@@ -0,0 +1,62 @@
+describe('csp-headers', () => {
+  // Currently unable to test spec based config values for experimentalCspAllowList
+  if (cy.config('experimentalCspAllowList') === false || cy.config('experimentalCspAllowList') === true) {
```

Comment:
This test looks conditional based on whatever `experimentalCspAllowList` is configured to. To get more coverage, we might want to move these to a `system-test`. That requires some domain knowledge so I can work on getting these moved over

Reply:
The original test was written with the idea that you could change the `experimentalCspAllowList` on a per spec basis. When I encountered https://github.com/cypress-io/cypress/issues/21151 I realized that the proxy would never get updated config values, so it became impossible to alter this value after initialization. I left the tests, but wrapped them in if statements as an example of how to handle both cases. If you know a way to integrate both through another means, that's preferred.

Reply:
the system test should allow us to integrate each case in this test to afford us more test coverage in testing this cases. When I get a chance I will push a commit to your branch refactoring this out into system tests

Reply:
I pushed e4780f82be to your most recent rebased branch which should refactor this out into system tests and more or less test the same behavior